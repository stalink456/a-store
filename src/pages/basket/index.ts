export { Basket as default } from './basket';
