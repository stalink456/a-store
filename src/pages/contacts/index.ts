export { Contacts as default } from './contacts';
