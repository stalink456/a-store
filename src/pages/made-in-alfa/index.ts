export { MadeInAlfa as default } from "./made-in-alfa";
