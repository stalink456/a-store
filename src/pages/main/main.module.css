@import "../../css/variables.css";

.main {
  display: flex;
  flex-wrap: nowrap;
  width: calc(100vw - 20px);
}

@media (max-width: 960px) {
  .main {
    flex-wrap: wrap;
    width: 100vw;
  }
}
