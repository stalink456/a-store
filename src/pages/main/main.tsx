import React from 'react';
import { MainPicture } from 'components/ui-components/main-picture';
import madeInAlfa from 'images/MadeInAlfa.jpeg';
import yourDesign from 'images/YourDesign.jpeg';

import styles from './main.module.css';

export const Main: React.FC = () => {
  return (
    <div className={styles.main}>
      <MainPicture
        src={madeInAlfa}
        text='Сделано в Альфе'
        route='/made-in-alfa'
      />
      <MainPicture src={yourDesign} text='Свой дизайн' route='/your-design' />
    </div>
  );
};
