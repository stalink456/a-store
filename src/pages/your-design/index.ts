export { YourDesign as default } from "./your-design";
