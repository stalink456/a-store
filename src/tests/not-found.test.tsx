import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import NotFound from 'pages/not-found';
import { Main } from 'pages/main';

describe('NotFound component', () => {
  it('should render NotFound component', () => {
    const { baseElement } = render(<NotFound />, { wrapper: MemoryRouter });

    expect(baseElement).toMatchSnapshot();
  });

  it('should render NotFound component with title', async () => {
    render(<NotFound />, { wrapper: MemoryRouter });

    await screen.findByText('Извините, что-то пошло не так');
  });

  it('should render NotFound component with a transition to Main page', async () => {
    render(
      <MemoryRouter initialEntries={['/not-found']}>
        <Routes>
          <Route path='/not-found' element={<NotFound />} />
          <Route path='/' element={<Main />} />
        </Routes>
      </MemoryRouter>
    );

    fireEvent.click(await screen.findByText('Главная страница'));

    expect(await screen.findByText('Свой дизайн')).toBeInTheDocument();
  });

  it("shouldn't render NotFound component with non-existent text", async () => {
    render(<NotFound />, { wrapper: MemoryRouter });

    expect(screen.queryByText('non-existent text')).not.toBeInTheDocument();
  });
});
