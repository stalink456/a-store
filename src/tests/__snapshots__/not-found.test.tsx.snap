// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render NotFound component 1`] = `
<body>
  <div>
    <div
      class="not_found"
    >
      <div
        class="space__spaceContainer_nr3as space__vertical_nr3as space__center_nr3as"
      >
        <div
          class="space__spaceItem_nr3as"
          style="margin-bottom: 20px;"
        >
          <div
            class="space__spaceContainer_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as not_found__text"
          >
            <div
              class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
            >
              <h1
                class="typography__component_19zgr typography__styrene-large_pxbut typography__medium_pxbut"
              >
                Извините, что-то пошло не так
              </h1>
            </div>
          </div>
        </div>
        <div
          class="space__spaceItem_nr3as"
        >
          <div
            class="space__spaceContainer_nr3as space__start_nr3as"
          >
            <div
              class="space__spaceItem_nr3as"
            >
              <button
                class="button__component_j4291 button__m_j4291 button__component_3xn9f button__secondary_3xn9f button__block_j4291"
                type="button"
              >
                <span
                  class="button__text_j4291 button__stretchText_j4291"
                >
                  Главная страница
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
